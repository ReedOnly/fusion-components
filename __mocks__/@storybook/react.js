import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
