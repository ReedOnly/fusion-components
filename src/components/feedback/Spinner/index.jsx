import React, { Children } from "react";
import classNames from "classnames";
import PropTypes from "prop-types";
import styles from "./styles.less";

const Spinner = ({
    floating,
    centered,
    primary,
    children,
    title,
    small,
    size,
    inline,
    ...props
}) => {
    const containerClassNames = classNames(styles.spinnerContainer, {
        [styles.floatingContainer]: floating,
        [styles.centeredContainer]: centered,
        [styles.inline]: inline,
    });

    const spinnerClassNames = classNames(styles.spinner, {
        [styles.primary]: primary,
        [styles.small]: small,
    });

    if (inline) {
        return (
            <div className={containerClassNames} {...props}>
                <div className={spinnerClassNames}>
                    <svg
                        className={styles.spinnerBox}
                        viewBox="24 24 48 48"
                        style={{ width: size, height: size }}
                    >
                        <circle
                            className={styles.spinnerBorder}
                            cx="48"
                            cy="48"
                            r="20"
                            fill="none"
                            strokeWidth={3}
                            strokeMiterlimit="10"
                        />
                    </svg>
                </div>
            </div>
        );
    }

    return (
        <div className={containerClassNames} {...props}>
            <div className={spinnerClassNames}>
                <svg className={styles.spinnerBox} viewBox="24 24 48 48">
                    <circle
                        className={styles.spinnerBorder}
                        cx="48"
                        cy="48"
                        r="20"
                        fill="none"
                        strokeWidth="3"
                        strokeMiterlimit="10"
                    />
                </svg>
            </div>
            {title ? <h2 className={styles.spinnerTitle}>{title}</h2> : null}
            {Children.count(children) > 0 ? <p>{children}</p> : null}
        </div>
    );
};

Spinner.propTypes = {
    /** Floats the spinner over content in the container */
    floating: PropTypes.bool,

    /** Center the spinner */
    centered: PropTypes.bool,

    /** Use primary color instead of default */
    primary: PropTypes.bool,

    /** Optional title */
    title: PropTypes.string,

    /** Use small spinner */
    small: PropTypes.bool,

    /** Inline the spinner */
    inline: PropTypes.bool,
};

Spinner.defaultProps = {
    floating: false,
    centered: false,
    primary: false,
    title: "",
    small: false,
    inline: false,
};

export default Spinner;
