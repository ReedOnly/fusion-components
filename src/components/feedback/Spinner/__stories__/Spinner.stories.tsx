import React from "react";
import { storiesOf } from "@storybook/react";
import Spinner from "..";

storiesOf("Feedback components/Spinner", module)
    .addParameters({ jest: ["Spinner.stories.jsx"] })
    .add("Default", () => (
        <React.Fragment>
            <Spinner />
            <Spinner primary />
            <Spinner small />
            <p>
                Inline spinner <Spinner inline />
            </p>
        </React.Fragment>
    ));
