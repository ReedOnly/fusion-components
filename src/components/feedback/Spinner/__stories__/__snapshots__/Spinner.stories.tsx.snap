// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback|Spinner Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      className="spinnerContainer"
    >
      <div
        className="spinner"
      >
        <svg
          className="spinnerBox"
          viewBox="24 24 48 48"
        >
          <circle
            className="spinnerBorder"
            cx="48"
            cy="48"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="3"
          />
        </svg>
      </div>
    </div>
    <div
      className="spinnerContainer"
    >
      <div
        className="spinner primary"
      >
        <svg
          className="spinnerBox"
          viewBox="24 24 48 48"
        >
          <circle
            className="spinnerBorder"
            cx="48"
            cy="48"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="3"
          />
        </svg>
      </div>
    </div>
    <div
      className="spinnerContainer"
    >
      <div
        className="spinner small"
      >
        <svg
          className="spinnerBox"
          viewBox="24 24 48 48"
        >
          <circle
            className="spinnerBorder"
            cx="48"
            cy="48"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="3"
          />
        </svg>
      </div>
    </div>
    <p>
      Inline spinner 
      <div
        className="spinnerContainer inline"
      >
        <div
          className="spinner"
        >
          <svg
            className="spinnerBox"
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
            viewBox="24 24 48 48"
          >
            <circle
              className="spinnerBorder"
              cx="48"
              cy="48"
              fill="none"
              r="20"
              strokeMiterlimit="10"
              strokeWidth={3}
            />
          </svg>
        </div>
      </div>
    </p>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
